src/Workspaces/Core/Portable/Classification/ClassificationContext.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.Classification
+{
+    /// <summary>
+    /// A context for aggregating classifications from one or more <see cref="ClassificationProvider"/>'s.
+    /// </summary>
+    internal class ClassificationContext
+    {
+        private readonly List<ClassifiedSpan> _spans;
```

Comment:
Yup -- otherwise we end up reallocating the internal array to create the ImmutableArray at the end.